pkg/frequencyplans/cflist.go

```diff
@@ -47,6 +47,27 @@ func CFList(fp FrequencyPlan, version ttnpb.PHYVersion) *ttnpb.CFList {
 	}
 }
 
+// TrimmedCFList generated by this frequency plan, for the version used by a device.
+// This function returns nil if the CFList could not be computed, or if the
+// device does not support CFLists. If the CFList contains more than 5 frequencies,
+// only the first 5 are used.
+func TrimmedCFList(fp FrequencyPlan, version ttnpb.PHYVersion) *ttnpb.CFList {
+	cfList := CFList(fp, version)
+	if cfList == nil {
+		return nil
+	}
+	switch cfList.Type {
+	case ttnpb.CFListType_FREQUENCIES:
```

Comment:
I think this is a perfect case for an if-statement, probably with early exit `return nil`